src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -289,9 +294,17 @@ public virtual VerifyResult VerifyForEachBlock(StoreView snapshot, BigInteger to
             return VerifyResult.Succeed;
         }
 
-        public virtual VerifyResult Verify(StoreView snapshot, BigInteger totalSenderFeeFromPool)
+        public virtual VerifyResult VerifyStateDependent(StoreView snapshot, BigInteger totalSenderFeeFromPool)
         {
-            VerifyResult result = VerifyForEachBlock(snapshot, totalSenderFeeFromPool);
+            BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, Sender);
+            BigInteger fee = SystemFee + NetworkFee + totalSenderFeeFromPool;
+            if (balance < fee) return VerifyResult.InsufficientFunds;
+            else return VerifyResult.Succeed;
+        }
+
+        public virtual VerifyResult VerifyStateIndependent(StoreView snapshot)
```

Comment:
If it's StateIndependent, why we need the `StoreView`?

Reply:
This is mainly due to applicationEngine requirements.
https://github.com/neo-project/neo/blob/0c94a678d7a152c0b8d90f45459cfe4a4e0a0bad/src/neo/SmartContract/ApplicationEngine.cs#L116-L120

But there are two syscalls, which also need snapshot.
- `System.Contract.IsStandard`
- `System.Runtime.CheckWitness`